src/color/p5.Color.js

```diff
@@ -446,52 +423,81 @@ var colorPatterns = {
  * </code>
  * </div>
  */
-p5.Color._getFormattedColor = function () {
+p5.Color._parseInputs = function() {
   var numArgs = arguments.length;
-  var mode    = this._renderer._colorMode;
-  var maxArr  = this._renderer._colorMaxes[this._renderer._colorMode];
+  var mode = this._renderer._colorMode;
+  var maxes = this._renderer._colorMaxes;
   var results = [];
```

Comment:
in these cases where we're creating a new variable to hold an empty array, I think it would be better to move this local variable outside of the function and reuse it. to empty it at the start you could just say `RGBA.length = 0;` I think this would be slightly better for performance.